### ISSUE TITLE
Set file sizing properties for iceberg tables?

### ISSUE THREAD

**Issue by nsivabalan** [apache/iceberg#4802]
May I know how can I set file sizing property while creating iceberg tables? I wish to configure max file size that gets created with iceberg. 

Options tried so far:

// lets say parquet_input_tbl refers to input table in parquet. 

1. spark.sql("CREATE TABLE s3.iceberg_1 USING iceberg partitioned by (VendorID) AS SELECT * FROM parquet_input_tbl order by VendorID").show()

this created files of sizes ~1Gb. 

2. spark.sql("CREATE TABLE s3.iceberg_2 USING iceberg partitioned by (VendorID) TBLPROPERTIES ('write.parquet.target-file-size-bytes '='52428800') AS SELECT * FROM parquet_input_tbl order by VendorID").show()

132945282 is the max file size.

From the [configs](https://iceberg.apache.org/docs/latest/configuration/), I deduced that parquet row group size matched above value. and so tried the next. 

3. spark.sql("CREATE TABLE s3.iceberg_3 USING iceberg partitioned by (VendorID) TBLPROPERTIES ('write.parquet.row-group-size-bytes'='52428800') AS SELECT * FROM parquet_input_tbl order by VendorID").show()

this resulted in 1GB files. 

so, may I know how to size the files using configs while creating iceberg tables.

**Comment by Zhangg7723** [comment]
The target size property in Iceberg is write.target-file-size-bytes

**Comment by ldwnt** [comment]
> The target size property in Iceberg is write.target-file-size-bytes

tried but it seems not working. the actual size of data file is beyond the value of parameter 'write.target-file-size-bytes'

ddl:
```
CREATE TABLE ...  (
...
) PARTITIONED BY (`dataset`) WITH (
    'format-version'='2',
    'engine.hive.enabled'='true',
    'write.metadata.delete-after-commit.enabled'='true',
    'write.metadata.previous-versions-max'='5',
    'write.target-file-size-bytes'='1024'
);
```

metadata file:
```
{
...
	"data_file": {
		"file_format": "PARQUET",
		"partition": {
			"dataset": {
				"string": "test.aa"
			}
		},
		"record_count": 1124,
		"file_size_in_bytes": 8663,
...
}
```

**Comment by RussellSpitzer** [comment]
The rolling file appended only checks the buffer size after writing a specific number of rows (to avoid the cost on every row). So for very small sizes are dominated by how many rows are checked before checking the size. This means setting a file size of 1kb or even less than 1meg will generally be inaccurate.

**Comment by idealspark** [comment]
I encountered the same error in version 0.12.1，it fixed in version 0.13.2。

in version 0.12.1，this code has bug
`
 # BaseTaskWriter
 private boolean shouldRollToNewFile() {
      return !format.equals(FileFormat.ORC) &&
          currentRows % ROWS_DIVISOR == 0 && 
          //check file should roll
          length(currentWriter) >= targetFileSize;  
    }
`


`
 //ParquetWriter,  shouldRollToNewFile code here
 @Override
  public long length() {
    try {
      if (closed) {
        return writer.getPos();
      } else {
        //this is a bug
        return writer.getPos() + (writeStore.isColumnFlushNeeded() ? writeStore.getBufferedSize() : 0);
      }
    } catch (IOException e) {
      throw new RuntimeIOException(e, "Failed to get file length");
    }
  }
`
